README.md
# TFG-IsometricMaps

LAStools/CHANGES.txt
14 September 2017 -- las2dem: fixed bug in elevation rasters when processing native compressed LAS 1.4 
5 September 2017 -- lasthin: new option keeps only points closest to a certain elevation '-percentile 40'
3 September 2017 -- lasgrid, las2dem, lasoverlap: option '-no_kml' avoids auto-creation of KML wrapper
28 August 2017 -- selected LAStools: fix issues when outputting LAS 1.4 as LAZ with "native LAS 1.4 extension"
27 August 2017 -- lassort, lasoptimize: fixed bug when sorting new LAS 1.4 point types by return number 
26 August 2017 -- lastrack: bug fix in new '-classify_xy_range_and_height_between' functionality
18 August 2017 -- NEW: lasoptimize prepares files for maximal compression and efficient spatial indexing
18 August 2017 -- all LAStools: support for Oblique Stereographic projection added for Volker Rail
18 August 2017 -- lastrack: new '-classify_xy_range_between' and 'classify_xy_range_and_height_between'
17 August 2017 -- LASlib: switch on "native LAS 1.4 extension". turns off with '-no_native'.
15 August 2017 -- lassort: fixed bug when sorting new LAS 1.4 point types by flight line (aka point source ID)
7 August 2017 -- lasclassify, lasnoise, lasthin: amount of console output controlled with '-verbose' or '-quiet'
5 August 2017 -- lassort, lasthin, lasduplicate: support '-remain_buffered' after '-buffered 25' to keep buffer
5 August 2017 -- LASlib: unless '-buffered 25' just created buffers always '-remain_buffered'
5 August 2017 -- LASlib: removed option '-unbuffered' because it makes too many assumptions
26 July 2017 -- las2tin: now supports DXF format as output option for the generated Delaunay TIN
18 July 2017 -- LASzip + LASlib: bug fix for spatially-indexed reading from native compressed LAS 1.4 files
14 July 2017 -- las2las: fixed missing 'comma' in compound (COMPD_CS) OGC WKT string 
13 July 2017 -- lasdiff: added missing checks for LAS 1.4 EVLR size and payloads
10 July 2017 -- lasground, lasground_new, lasheight: amount of console output controlled with '-verbose' or '-quiet'
5 July 2017 -- lasreturn: option '-repair_number_of_returns' now also works for new LAS 1.4 point types 6 to 10
28 June 2017 -- lasgrid, las2dem, blast2dem, lascanopy: fixed quantization bug in X/Y coords for LAZ grid output
28 June 2017 -- lasgrid, las2dem, blast2dem, lascanopy: no more bug for unlicensed writing (black-diagonal) 
25 June 2017 -- lascolor: fix inability to read a certain type of tiled 4-band TIF images
24 June 2017 -- lasgrid, las2dem, lascanopy: correct *unsigned* output for 16 bit '-intensity' for '-no_data 0'
24 June 2017 -- LASlib: corrected on-the-fly conversion from BIL rasters to point clouds for unsigned integers
23 June 2017 -- lasnoise: fixed that '-ignore_class 8' did not ignore points during isolated computation
22 June 2017 -- laslayers: added ability to run on multiple '-cores 7'
18 June 2017 -- lasboundary + all GUIs: fixed "small polygons missing bug" in '-use_lax' mode and in GUI previews
8 June 2017 -- lasclassify: IMPORTANT BUG FIX (but only if you downloaded version 170528 where it was introduced)
28 May 2017 -- LASzip DLL: support for "selective decompression" of compressed LAS 1.4 points added into DLL API
27 May 2017 -- lascontrol: let '-odir quality' specify where error report '-cp_out cp_check.csv' will be stored 
21 May 2017 -- lasgrid: fixed BIL output for the '-point_density' option. now '-density' means '-point_density'.
13 May 2017 -- all (license-restricted) LAStools: option '-fail' prevents running when license is missing/expired
11 May 2017 -- LASlib: new transforms '-copy_classification_into_user_data' and '-copy_attribute_into_user_data 1'
10 May 2017 -- las2dem, blast2dem, lasgrid: fix rounding error for ususual or very small step sizes (0.2 or less)
5 May 2017 -- lasground + lasground_new: user data change due to '-compute_height' makes it into LASlayers output
28 April 2017 -- lasboundary: allow creation of KML and SHP '-overview' files for multiple LAS/LAZ files
26 April 2017 -- lasboundary: also adding option to add '-labels' to SHP output that populate the DBF attibutes
25 April 2017 -- LASzip DLL: "native LAS 1.4 extension" in LASzip DLL via 'laszip_request_native_extension()'
24 April 2017 -- lastile: also '-remove_buffer' for buffers generated by on-the-fly buffering (= '-buffered 50')
22 April 2017 -- lasground and lasground_new: fix of bb extend for '-buffered 20' and '-remain_buffered' combo
20 April 2017 -- LASlib: new option '-unbuffered' for lasreadopener removes buffers from on-the-fly buffering
16 April 2017 -- txt2las: allow requesting a particular point type with new option '-set_point_type 6'
14 April 2017 -- LASlib: enable "adaptive chunking" for "native LAS 1.4 extension" for compressing new LAS 1.4
8 April 2017 -- LASlib: new check to detect corrupt LAZ files generated by libLAS for files with "Extra Bytes"
30 March 2017 -- LASlib: alpha-release of "native LAS 1.4 extension" for LASzip compression of new LAS 1.4 points
29 March 2017 -- LASlib: enable "native LAS 1.4 extension" for LASzip via '-native'
27 March 2017 -- lascolor: support NIR, 4 band RGBNIR images as well as three band grey and single band copying
26 March 2017 -- lascanopy: fixed bug in raster mode when attempting to use '-use_tile_bb' or '-use_orig_bb'
22 March 2017 -- las2shp: fixed bug accidentally introduced in version 170203
22 March 2017 -- lassort: fixed *serious* bug accidentally introduced in version 170313
22 March 2017 -- LASlib: new filter '-thin_points_with_time 0.000001' vs '-thin_pulses_with_time 0.001'
12 March 2017 -- LASlib: new filters '-drop_every_nth' '-drop_number_of_returns' '-keep_number_of_returns'
11 March 2017 -- lassort: option to first sort by '-scanner_channel' and then by '-return_number' 
11 March 2017 -- lassort: option to first sort by '-point_source' and then by '-scanner_channel'

src/script3.py
import os, sys

minx = 13
maxx = 14
miny = 15
maxy = 16

#minx = 36
#maxx = 104
#miny = 32
#maxy = 100
#minx = 18
#maxx = 52
#miny = 16
#maxy = 50
#minx = 9
#maxx = 26
#miny = 8
#maxy = 25

# Progress bar
toolbar_width = maxx - minx + 1

# setup toolbar
sys.stdout.write("[%s]" % (" " * toolbar_width))
sys.stdout.flush()
sys.stdout.write("\b" * (toolbar_width+1)) # return to start of line, after '[' 

x = minx
while x <= maxx:
	sys.stdout.write("=")
	sys.stdout.flush()
	y = miny
	while y <= maxy:		
		actualx = x * 2
		actualy = y * 2

		if os.path.isfile('/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx) + '_' + str(actualy) + '.png'):
			file1 = '/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx) + '_' + str(actualy) + '.png'
		else:
			file1 = 'negro.png'
		if os.path.isfile('/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx) + '_' + str(actualy + 1) + '.png'):
			file2 = '/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx) + '_' + str(actualy + 1) + '.png'
		else:
			file2 = 'negro.png'
		if os.path.isfile('/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx + 1) + '_' + str(actualy) + '.png'):
			file3 = '/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx + 1) + '_' + str(actualy) + '.png'
		else:
			file3 = 'negro.png'
		if os.path.isfile('/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx + 1) + '_' + str(actualy + 1) + '.png'):
			file4 = '/media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/6/map_' + str(actualx + 1) + '_' + str(actualy + 1) + '.png'
		else:
			file4 = 'negro.png'			

		os.system('convert -append ' + file1 + ' ' + file2 + ' out1.png')
		os.system('convert -append ' + file3 + ' ' + file4 + ' out2.png')
		os.system('convert +append out1.png out2.png out.png')
		os.system('convert out.png -resize 256x181\! /media/pablo/280F8D1D0A5B8545/TFG_files/cliente_local/45/N/5/map_' + str(int(actualx / 2)) + '_' + str(int(actualy / 2)) + '.png')

		y += 1
	x += 1

os.system('rm out1.png out2.png out.png')
sys.stdout.write("\n")
print('DONE')			